calcrule_third_party_payment/utils.py

```diff
@@ -11,3 +18,143 @@ def check_bill_exist(instance, convert_to, **kwargs):
             bills = BillItem.objects.filter(line_type=content_type, line_id=claim.id)
             if bills.count() == 0:
                 return True
+
+
+def claim_batch_valuation(work_data):
+    """ update the service and item valuated amount """
+    allocated_contributions = work_data["allocated_contributions"]
+    allocated_contributions_ip = work_data["allocated_contributions_ip"] if "allocated_contributions_ip" in work_data else None
+    relative_price_g = work_data["relative_price_g"]
+    product = work_data["product"]
+    items = work_data["items"]
+    services = work_data["services"]
+    start_date = work_data["start_date"]
+    start_date_ip = work_data["start_date_ip"]
+    end_date = work_data["end_date"]
+    claims = work_data["claims"]
+
+    # Sum up all item and service amount
+    value_hospital = 0
+    value_non_hospital = 0
+    index = 0
+    index_ip = 0
+
+    # if there is no configuration the relative index will be set to 100 %
+    if start_date is not None or start_date_ip is not None:
+        claims = add_hospital_claim_date_filter(claims, relative_price_g, start_date, start_date_ip, end_date,
+                                                product.ceiling_interpretation)
+        claims = claims.prefetch_related(
+            Prefetch('items', queryset=ClaimItem.objects.filter(legacy_id__isnull=True).filter(product=product))
+        ).prefetch_related(
+            Prefetch('services', queryset=ClaimService.objects.filter(legacy_id__isnull=True).filter(product=product))
+        )
+        claims = list(claims.values_list('id', flat=True).distinct())
+        claims = Claim.objects.filter(id__in=claims)
```

Comment:
I don't get that line
```
claims = list(claims.values_list('id', flat=True).distinct())
claims = Claim.objects.filter(id__in=claims)
```
Normally the claims queryset will give you distinct claims  so the two line above seems useless but I can be misstaken